src/SolutionNormPostProcessing.C

```diff
@@ -339,6 +339,12 @@ SolutionNormPostProcessing::execute()
     int offSet = 0;
     for ( size_t j = 0; j < fieldPairVec_.size(); ++j ) {
 
+      // sync fields
+      fieldPairVec_[j].first->modify_on_device();
```

Comment:
@ashesh2512 These should be marked as modified where they were updated. Only need `sync_to_host` here.

Reply:
@sayerhs Thanks for pointing that out. I was trying to be safe, but I have pushed a commit removing the redundant calls - https://github.com/Exawind/nalu-wind/pull/838/commits/4911b48139317aaee25402b59baa3fbb4a35a4e6